packages/compiler-sfc/src/template/transformAssetUrl.ts

```diff
@@ -168,7 +168,13 @@ function getImportsExpressionExp(
         loc,
         ConstantTypes.CAN_STRINGIFY
       )
-      context.imports.push({ exp, path })
+      // When path starts with /, for example, /foo/bar.png, it is an absolute file path
+      // In this case we need to ensure the path is not encoded (to %2F),
```

Comment:
Not only does the bug exist for absolute paths, but relative imports would fail, too:
https://stackblitz.com/edit/vitejs-vite-aq7cki?file=src%2FApp.vue&startScript=build